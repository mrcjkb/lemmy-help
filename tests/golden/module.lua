---@mod mod.intro Introduction
---@brief [[
---
---We can have multiple `---@mod` tags so that we can have a block only for text.
---This is for the cases where you want bunch of block only just for text
---and does not contains any code.
---
---@brief ]]

---@mod mod.Human Human module

local U = {}

---The Homosapien
---@class Human
---@field legs number Total number of legs
---@field hands number Total number of hands
---@field brain boolean Does humans have brain?

---Default traits of a human
---@type Human
U.DEFAULT = {
    legs = 2,
    hands = 2,
    brain = false,
}

---Creates a Human
---@return Human
---@usage [[
---local H = require('Human')
---local human = H:create()
---
---print(getmetatable(human))
---@usage ]]
function U:create()
    return setmetatable(self.DEFAULT, { __index = self })
end

return U
