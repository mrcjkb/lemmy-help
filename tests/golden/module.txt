==============================================================================
Introduction                                                         *mod.intro*


We can have multiple `---@mod` tags so that we can have a block only for text.
This is for the cases where you want bunch of block only just for text
and does not contains any code.


==============================================================================
Human module                                                         *mod.Human*

Human                                                                    *Human*
    The Homosapien

    Fields: ~
        {legs}   (number)   Total number of legs
        {hands}  (number)   Total number of hands
        {brain}  (boolean)  Does humans have brain?


U.DEFAULT                                                            *U.DEFAULT*
    Default traits of a human

    Type: ~
        (Human)


U:create()                                                            *U:create*
    Creates a Human

    Returns: ~
        (Human)

    Usage: ~
>lua
        local H = require('Human')
        local human = H:create()

        print(getmetatable(human))
<



